packages/fxa-payments-server/src/lib/apiClient.ts

```diff
@@ -126,11 +127,13 @@ export async function apiUpdateSubscriptionPlan(params: {
   subscriptionId: string;
   planId: string;
   productId: string;
+  paymentProvider: ProviderType | undefined;
```

Comment:
If I understand the question correctly, all the "old" customer should have a payment provider since that is now included in the customer returned by the backend.

Reply:
Thanks; in that case, the tricksy part here is that our `Customer` type defines `payment_provider` as an optional property due to the `SubscriptionCreate` scenario. In which case I get TS errors if I assume that `paymentProvider` is just `ProviderType` e.g. in `SubscriptionUpgrade`:

```
 const paymentProvider: ProviderType = customer.payment_provider;
```

Error:
```
Type 'ProviderType | undefined' is not assignable to type 'ProviderType'.
  Type 'undefined' is not assignable to type 'ProviderType'.ts(2322)
```

Do you feel strongly one way or the other? (Leaving the `ProviderType | undefined` to avoid TS errors or being more precise in the correct scenarios and specifying `ProviderType` only even if it means a TS error)

Reply:
Oh sorry, yeah, we can leave it as `ProviderType | undefined`.